services/rfq/relayer/pricer/fee_pricer.go

```diff
@@ -147,11 +157,95 @@ func (f *feePricer) GetDestinationFee(parentCtx context.Context, _, destination
 		fee = new(big.Int).Add(fee, l1Fee)
 		span.SetAttributes(attribute.String("l1_fee", l1Fee.String()))
 	}
+
+	// If specified, calculate and add the call fee, as well as the call value which will be paid by the relayer
+	if tx != nil {
+		fee, err = f.addZapFees(ctx, destination, denomToken, isQuote, tx, fee)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	span.SetAttributes(attribute.String("destination_fee", fee.String()))
 	return fee, nil
 }
 
-func (f *feePricer) GetTotalFee(parentCtx context.Context, origin, destination uint32, denomToken string, isQuote bool) (_ *big.Int, err error) {
+//nolint:gosec
+func (f *feePricer) addZapFees(ctx context.Context, destination uint32, denomToken string, isQuote bool, tx *fastbridgev2.IFastBridgeV2BridgeTransactionV2, fee *big.Int) (*big.Int, error) {
+	span := trace.SpanFromContext(ctx)
+
+	if len(tx.CallParams) > 0 {
+		gasEstimate, err := f.getZapGasEstimate(ctx, destination, tx)
+		if err != nil {
+			return nil, err
+		}
+		callFee, err := f.getFee(ctx, destination, destination, int(gasEstimate), denomToken, isQuote)
+		if err != nil {
+			return nil, err
+		}
+		fee = new(big.Int).Add(fee, callFee)
+		span.SetAttributes(attribute.String("call_fee", callFee.String()))
+	}
+
+	if tx.CallValue != nil && tx.CallValue.Sign() > 0 {
+		callValueFloat := new(big.Float).SetInt(tx.CallValue)
+		valueDenom, err := f.getDenomFee(ctx, destination, destination, denomToken, callValueFloat)
+		if err != nil {
+			return nil, err
+		}
+		valueScaled, err := f.getFeeWithMultiplier(ctx, destination, isQuote, valueDenom)
```

Comment:
@dwasse I think this will be using `relay_fixed_fee_multiplier` when checking whether to relay a submitted user transaction. This value is under `1.0`, and therefore opens up a griefing vector (e.g. ask for  `CallValue` worth $1, but only supply $0.8 in fees)

Reply:
I think one of these two approaches should work here:
1. Always use `quote_fixed_fee_miltiplier` for determining how much is the requested `CallValue` worth
2. Introduce a separate multiplier parameter for that